optuna/study/_optimize.py

```diff
@@ -193,7 +193,7 @@ def _run_trial(
 
     state: Optional[TrialState] = None
     values: Optional[List[float]] = None
-    func_err: Optional[Exception] = None
+    func_err: Optional[BaseException] = None
```

Comment:
I personally prefer `Union[Exception,KeyboardInterrupt]` over `BaseException`, because `BaseException` is little bit too general.